File: javascript/1422.Maximum-Score-After-Splitting-a-String.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function (s) {
  let score = 0;
  let ones = 0;
  let zeros = 0;
  let index = 0;

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "1") ones += 1;
  }

  while (index < s.length - 1) {
    if (s[index] === "0") {
      zeros += 1;
    } else {
      ones -= 1;
    }

    score = Math.max(score, zeros + ones);

    index += 1;
  }

  return score;
};

File: python/1422.Maximum-Score-After-Splitting-a-String.py
class Solution:
    def maxScore(self, s: str) -> int:
        ones, zeros, score = 0, 0, 0

        for c in s:
            if c == "1":
                ones += 1

        for i in range(len(s) - 1):
            c = s[i]
            if c == "0":
                zeros += 1
            else:
                ones -= 1
            score = max(score, ones + zeros)

        return score